src/nvim/mouse.c

```diff
@@ -681,7 +681,11 @@ bool do_mouse(oparg_T *oap, int c, int dir, int count, bool fixindent)
     if (in_statuscol && wp->w_p_rl) {
       click_col = wp->w_width_inner - click_col - 1;
     }
-    if (in_statuscol && click_col >= (int)wp->w_statuscol_click_defs_size) {
+
+    if ((in_statuscol && click_col >= (int)wp->w_statuscol_click_defs_size)
+        || (in_status_line
+            && click_col >=
+            (int)(global_stl_height() > 0 ? curwin : wp)->w_status_click_defs_size)) {
```

Comment:
`global_stl_height() > 0` is used multiple times in this function. Assign it to a variable?